app/post_metadata.go

```diff
@@ -346,17 +347,19 @@ func (a *App) getLinkMetadata(requestURL string, timestamp int64, isNewPost bool
 
 	request.Header.Add("Accept", "text/html, image/*")
 
-	res, err := a.HTTPService.MakeClient(false).Do(request)
-	if err != nil {
-		return nil, nil, err
-	}
+	client := a.HTTPService.MakeClient(false)
+	client.Timeout = time.Duration(*a.Config().ExperimentalSettings.LinkMetadataTimeout) * time.Millisecond
 
-	defer res.Body.Close()
+	res, err := client.Do(request)
 
-	// Parse the data
-	og, image, err = a.parseLinkMetadata(requestURL, res.Body, res.Header.Get("Content-Type"))
+	if err == nil {
+		defer res.Body.Close()
+
+		// Parse the data
+		og, image, err = a.parseLinkMetadata(requestURL, res.Body, res.Header.Get("Content-Type"))
+	}
 
-	// Write back to cache and database
+	// Write back to cache and database, even if there was an error and the results are nil
```

Comment:
Am I correct that the cacheing and storing of nil results are to prevent further requests of the same result? What about the error cases?

Reply:
That's correct, and it's intended that we cache that for error requests as well. Elias and I spent some time discussing that problem earlier, and we came to the conclusion that adding retry logic doesn't make sense in the majority of cases since most of them would be permanent issues like MM being on an airgapped network or the URL being incorrect. Network instability is the only time I think it'd really be valuable, but since this is on the server side of things, it's much less likely that the server has a poor network connection compared to the apps.